### PR TITLE
Add regression tests for rdar://58495602

### PR DESCRIPTION
**Pull request by CodaFi**

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please smoke test and merge

**Comment by CodaFi** [comment]
@swift-ci please clean smoke test macOS platform

File: test/SILOptimizer/di_property_wrappers_errors_multifile.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// rdar://58495602: The order used to matter here. If use compiles before def
+// then you got a DI error. If def compiles before use then you got a linker error.
+
+// RUN: %target-swift-frontend -emit-sil -verify %s %S/Inputs/di_property_wrappers_errors_multifile_2.swift
+// RUN: %target-swift-frontend -emit-sil -verify %S/Inputs/di_property_wrappers_errors_multifile_2.swift %s
+
+let contentView = ContentView() // expected-error {{missing argument for parameter 'runner' in call}}
```
Comment (slavapestov) [inline test/SILOptimizer/di_property_wrappers_errors_multifile.swift]: Is this really a DI test? It looks like the error comes from Sema to me.

Comment (CodaFi) [reply test/SILOptimizer/di_property_wrappers_errors_multifile.swift]: The error comes from sema, but for a regression to occurs we must run DI.

**Comment by CodaFi** [comment]
@swift-ci please clean smoke test macOS platform

**Comment by CodaFi** [comment]
Still blocked on sourcekitd failures...

**Comment by CodaFi** [comment]
@swift-ci please smoke test macOS platform

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**